templates/class.mustache

```diff
@@ -75,7 +80,7 @@ export class {{&className}} {
         this.configureRequestHandler = handler;
     }
 
-    private request(method: string, url: string, body: any, headers: RequestHeaders, queryParameters: any, form: any, reject: CallbackHandler, resolve: CallbackHandler) {
+    private request(method: string, url: string, body: any, headers: RequestHeaders, queryParameters: QueryParameters, form: any, reject: CallbackHandler, resolve: CallbackHandler, opts: CommonRequestOptions) {
```

Comment:
tighter type checks (`queryParams`) and add `opts` for `retries`, `timeout`, and `deadline` support